providers/curation/github.js

```diff
@@ -173,51 +173,29 @@ class GitHubCurationService {
 
   // Return an array of valid patches that exist
   // and a list of definitions that do not exist in the store
-  async _validateComponentDefinitionExists(patches) {
-    const coordinatesWithRevisions = flatten(
-      map(patches, patch =>
-        map(patch.revisions, (_, key) => {
-          return Object.assign({}, patch.coordinates, {
-            revision: key
-          })
-        })
-      )
-    )
-
-    // Check if all the definitions exists into the store
-    const validDefinitions = await this.definitionService.listAll(coordinatesWithRevisions)
-    if (coordinatesWithRevisions.length !== validDefinitions.length) {
-      const notFoundDefinitions = []
-      map(patches, patch =>
-        map(
-          patch.revisions,
-          throat(1, async (_, key) => {
-            const coordinates = Object.assign({}, patch.coordinates, {
-              revision: key
-            })
-            if (validDefinitions.findIndex(definition => isEqual(definition, coordinates)) === -1) {
-              notFoundDefinitions.push(
-                `Definition ${EntityCoordinates.fromObject(coordinates).toString()} has not been found`
-              )
-            }
-          })
+  async _validateDefinitionsExist(patches) {
+    const targetCoordinates = patches.reduce((result, patch) => {
```

Comment:
Jeff suggested to use this code, but it was wrong so i've updated it in order to work properly:

 ```
const targetCoordinates = patches.reduce((patch, result) => {
      for (let key in patch.revisions)
        result.push(EntityCoordinates.fromObject({ ...patch.coordinates, revision: key }))
    }, [])
    const validDefinitions = await this.definitionService.listAll(targetCoordinates)
    return targetCoordinates.reduce(
      (coordinates, result) => (result[validDefinitions.includes(coordinates) ? 'valid' : 'missing']).push(coordinates),
      { valid: [], missing: [] }
    )
```